### PR TITLE
WIP: Refactoring: Parameterize constants with role names

### PR DESCRIPTION
**Pull request by seintur**
Associate constants with role names to factorize the code of the forName factory method.

### CODE REVIEW

**Comment by pvojtechovsky** [comment]
By the way, why we need synonyms for role names? Backward comaptibility? More friendly for clients? If there is not goodanswer for that, then lets remove these synonyms.

File: src/main/java/spoon/reflect/path/CtRole.java [hunk 1]
```diff
@@ -59,25 +63,31 @@
...
 	ARGUMENT,
-	SUPER_TYPE,
+	SUPER_TYPE("superclass"),
 	NESTED_TYPE,
 	CONSTRUCTOR,
 	EXECUTABLE,
-	FIELD,
-	CAST,
-	VALUE,
+	FIELD("typemembers"),
```
Comment (pvojtechovsky) [inline src/main/java/spoon/reflect/path/CtRole.java]: executables and nested types are type members too ... it probably needs further refactoring ... but after tjis PR.

**Comment by monperrus** [comment]
> If there is not goodanswer for that, then lets remove these synonyms.
I agree. We should remove them. CtPath should simply use Rol.tolowercase().

This is not backward compatible but AFAIK nobody uses CtPath so far (it is not comprehensive  and
too experimental).

**Comment by surli** [comment]
> We should remove them. CtPath should simply use Rol.tolowercase().

Does this mean we should close this PR? I fixed the conflict in case.

**Comment by monperrus** [comment]
closed per #1702 

thanks @seintur for bringing this on the table.

**Pull request closed without merging**